bigchaindb/models.py

```diff
@@ -0,0 +1,209 @@
+from functools import reduce
+from operator import and_
+
+from bigchaindb_common.crypto import hash_data, VerifyingKey, SigningKey
+from bigchaindb_common.exceptions import (InvalidHash, InvalidSignature,
+                                          OperationError, DoubleSpend,
+                                          TransactionDoesNotExist)
+from bigchaindb_common.transaction import Transaction
+from bigchaindb_common.util import gen_timestamp, serialize
+
+
+class Transaction(Transaction):
+    def validate(self, bigchain):
+        """Validate a transaction.
+
+        Args:
+            bigchain (Bigchain): an instantiated bigchaindb.Bigchain object.
+
+        Returns:
+            The transaction (Transaction) if the transaction is valid else it
+            raises an exception describing the reason why the transaction is
+            invalid.
+
+        Raises:
+            OperationError: if the transaction operation is not supported
+            TransactionDoesNotExist: if the input of the transaction is not found
+            TransactionOwnerError: if the new transaction is using an input it doesn't own
+            DoubleSpend: if the transaction is a double spend
+            InvalidHash: if the hash of the transaction is wrong
+            InvalidSignature: if the signature of the transaction is wrong
+        """
+        if len(self.fulfillments) == 0:
+            raise ValueError('Transaction contains no fulfillments')
+
+        input_conditions = []
+        inputs_defined = reduce(and_, [bool(ffill.tx_input) for ffill
+                                       in self.fulfillments])
```

Comment:
A better way to do this is `inputs_defined = all([ffill.tx_input for ffill in self.fulfillments)]`